src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -117,6 +117,9 @@ class SemanticOccurenceAtPositionRequest extends OccurenceAtPositionRequest {
 	}
 
 	protected _compute(model: ITextModel, selection: Selection, wordSeparators: string, token: CancellationToken): Promise<DocumentHighlight[]> {
+		if (!selection.isEmpty()) {
+			return Promise.resolve([]);
+		}
 		return getOccurrencesAtPosition(this._providers, model, selection.getPosition(), token).then(value => value || []);
```

Comment:
Could we do this behind a setting? We could then set up an experiment and try to change the default and see what kind of feedback comes in.